Public/Src/Engine/Scheduler/Fingerprints/PipFingerprinter.cs

```diff
@@ -244,7 +246,17 @@ protected virtual void AddWeakFingerprint(IFingerprinter fingerprinter, SealDire
         /// </summary>
         protected virtual void AddWeakFingerprint(IFingerprinter fingerprinter, Process process)
         {
-            AddFileDependency(fingerprinter, "Executable", process.Executable);
+            if (process.UntrackedPaths.Contains(process.Executable.Path) ||

```

Comment:
Add some unit tests as well. See `PipFingerprinterTests`

Reply:
It would also be good to add an integration test.  See [`ChangeAffectedInputTests`](https://github.com/microsoft/BuildXL/blob/master/Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/ChangeAffectedInputTests.cs); add a new test method at the end of that class which: 
  - sets up a simple process pip (e.g., one that has a single `WriteFile` operation)
  - runs the scheduler and asserts the pip was executed (i.e., was a cache miss) and succeeded
  - changes the executable of that process pip
  - runs the scheduler again and asserts that the pip was was a cache hit
  - reverts the change it made to the executable of that process pip